src/Chapter4.hs

```diff
@@ -489,6 +497,27 @@ Implement the 'Applicative' instance for our 'List' type.
   type.
 -}
 
+concatenateList :: List a -> List a -> List a
+concatenateList Empty b = b
```

Comment:
Perfect!

You can one more case to make your solution even more efficient:

```haskell
concatenateList a Empty = a
```

In that case, you won't need to walk through each element of the first list when the second one is empty and return the list instantly 🚤

Reply:
Done =)